common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -119,15 +115,24 @@ public static UTF8String blankString(int length) {
     return fromBytes(spaces);
   }
 
-  protected UTF8String(Object base, long offset, int numBytes) {
+  protected UTF8String(byte[] bytes, long offset, int numBytes) {
+    this(new ByteArrayMemoryBlock(bytes, offset, numBytes));
+  }
+
+  public UTF8String(MemoryBlock base) {
     this.base = base;
-    this.offset = offset;
-    this.numBytes = numBytes;
+    if (base != null) {
```

Comment:
I think we can actually add ab assert to make sure `base` is not null.

Reply:
`UTF8String(null)` is actually called for serialization. Thus, we cannot replace `if` with `assert`.

Reply:
ah i see